pkg/kn/core/root.go

```diff
@@ -234,6 +235,15 @@ func initConfigFlags() {
 	var aBool bool
 	aBool = viper.GetBool("lookup-plugins")
 	commands.Cfg.LookupPlugins = &aBool
+
+	// set the Cfg.SinkPrefixes from viper if sink is configured
+	if viper.IsSet("sink") {
+		err := viper.UnmarshalKey("sink", &commands.Cfg.SinkPrefixes)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Unable to parse sink prefixes configuration: %v .\n", err)
```

Comment:
can we propagate this up to the top level (i.e. let this method return an error), and then stop ? Otherwise it would just continue, but I think we should deal this misconfiguration as an error. If possible, you should also add the configuration file (could maybe be obtained from `viper`) and the cake on the ice would be the line number (if it's easy to detect).

Reply:
raise an error in the new version. The error message returned from `viper.UnmarshalKey` is included in the final error message, together with the config file information.